presto-main/src/test/java/com/facebook/presto/sql/TestExpressionInterpreter.java

```diff
@@ -686,6 +686,8 @@ public void testCastOptimization()
         assertOptimizedEquals("cast(bound_integer + 1 as VARCHAR)", "'1235'");
         assertOptimizedEquals("cast(bound_long + 1 as VARCHAR)", "'1235'");
         assertOptimizedEquals("cast(unbound_string as VARCHAR)", "cast(unbound_string as VARCHAR)");
```

Comment:
why this didn't rewrite?

Reply:
That looked odd to me, too. Apparently, assertOptimizedEquals checks that the result of optimizing the left is the same as the result of optimizing the right. I'm not sure why that test is even there. Something we'll have to look at in more depth and clean up later.